tasks.php

```diff
@@ -340,22 +339,31 @@ function SearchParams_echo_controls()
 // initializing it with any (valid) value that the current request
 // has supplied for the parameter.
 {
-    global $SearchParams_choices;
+    global $SearchParams_choices, $tasks_url;
 
     if (isset($_REQUEST['search_text']) && !empty($_REQUEST['search_text'])) {
         $st = $_REQUEST['search_text'];
         $search_text = attr_safe($st);
     }
     else $search_text = "";
 
+    echo "<form action='$tasks_url' method='get'>";
+    echo "<table class='themed'>\n";
+    echo "<tr><th>" . _("Search") . "</th>\n";
+    echo "<td>";
+
     echo "<input type='hidden' name='action' value='search'>\n";
-    echo "<input type='text' value='$search_text' name='search_text' size='50'>\n";
+    echo "<input type='text' value='$search_text' name='search_text' style='width: 20em'>\n";
 
     foreach ($SearchParams_choices as $param_name => $choices)
     {
         $value = (int) get_enumerated_param($_REQUEST, $param_name, '999', array_keys($choices));
         echo dropdown_select($param_name, $value, $choices);
     }
+
+    echo "<input type='submit' value='Search'></td>\n";
```

Comment:
Localization? I see we do it sometimes and not others, so probably something to address later.

Reply:
Very little of this file has been localized, unfortunately. I did some in places I touched when I was thinking about it -- but clearly not all 😦 . It's probably worth a separate MR to clean all of these up (and a second separate MR to update all of the `mysqli_query()` and escape clauses.